crates/block-producer/src/rpc_client.rs

```diff
@@ -120,6 +143,120 @@ impl RPCClient {
         Ok(None)
     }
 
+    /// this function queries identity cell by args
+    pub async fn query_identity_cell(&self, args: Bytes) -> Result<Option<CellInfo>> {
+        let search_key = SearchKey {
+            script: ckb_types::packed::Script::new_builder()
+                .code_hash(ckb_types::prelude::Pack::pack(&*TYPE_ID_CODE_HASH))
+                .hash_type(ScriptHashType::Type.into())
+                .args(ckb_types::prelude::Pack::pack(&args))
+                .build()
+                .into(),
+            script_type: ScriptType::Type,
+            filter: None,
+        };
+        let order = Order::Desc;
+        let limit = Uint32::from(DEFAULT_QUERY_LIMIT as u32);
+
+        let mut cell = None;
+        let mut cursor = None;
+        while cell.is_none() {
+            let cells: Pagination<Cell> = to_result(
+                self.indexer_client
+                    .request(
+                        "get_cells",
+                        Some(ClientParams::Array(vec![
+                            json!(search_key),
+                            json!(order),
+                            json!(limit),
+                            json!(cursor),
+                        ])),
+                    )
+                    .await?,
+            )?;
+            cursor = Some(cells.last_cursor);
+            assert!(
+                cells.objects.len() <= 1,
+                "Never returns more than 1 identity cells"
+            );
+            cell = cells.objects.into_iter().find_map(|cell| {
+                // delete cells with data & type
+                if !cell.output_data.is_empty() || cell.output.type_.is_some() {
```

Comment:
Why filter out cells with data and type? I think target identity cell must have data and type script.

Reply:
You are right, this is a copy-paste error.

Reply:
I will close this PR please use @EthanYuan 's fork